src/icon-layer.ts

```diff
@@ -0,0 +1,209 @@
+import {
+  CompositeLayer,
+  CompositeLayerProps,
+  DefaultProps,
+  GetPickingInfoParams,
+  Layer,
+  LayersList,
+  assert,
+} from "@deck.gl/core";
+import { IconLayer } from "@deck.gl/layers";
+import type { IconLayerProps } from "@deck.gl/layers";
+import * as arrow from "apache-arrow";
+import * as ga from "@geoarrow/geoarrow-js";
+import {
+  assignAccessor,
+  extractAccessorsFromProps,
+  getGeometryVector,
+  invertOffsets,
+} from "./utils.js";
+import {
+  GeoArrowExtraPickingProps,
+  computeChunkOffsets,
+  getPickingInfo,
+} from "./picking.js";
+import { ColorAccessor, FloatAccessor, GeoArrowPickingInfo } from "./types.js";
+import { EXTENSION_NAME } from "./constants.js";
+import { validateAccessors } from "./validate.js";
+
+/** All properties supported by GeoArrowScatterplotLayer */
+export type GeoArrowIconLayerProps = Omit<
+  IconLayerProps<arrow.Table>,
+  | "data"
+  | "getPosition"
+  | "getRadius"
+  | "getFillColor"
+  | "getLineColor"
+  | "iconAtlas"
+  | "iconMapping"
+> &
+  _GeoArrowIconLayerPropsProps &
+  CompositeLayerProps;
+
+/** Properties added by GeoArrowScatterplotLayer */
+type _GeoArrowIconLayerPropsProps = {
+  data: arrow.Table;
+
+  /**
+   * If `true`, validate the arrays provided (e.g. chunk lengths)
+   * @default true
+   */
+  _validate?: boolean;
+  /**
+   * Center position accessor.
+   * If not provided, will be inferred by finding a column with extension type
+   * `"geoarrow.point"` or `"geoarrow.multipoint"`.
+   */
+  getPosition?: ga.vector.PointVector | ga.vector.MultiPointVector;
```

Comment:
For now, you should remove `| ga.vector.MultiPointVector` since this layer as written only supports points